src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -483,6 +483,112 @@ private IOperation CreateBoundUnconvertedAddressOfOperatorOperation(BoundUnconve
+                    {
+                        data = null;
+                        return false;
+                    }
+
+                    attributedDeclaration = baseFieldDeclarationSyntax.Declaration.Variables[0];
+                }
+
+                var model = _semanticModel.Compilation.GetSemanticModel(boundAttribute.SyntaxTree!);
+                attributedSymbol = model.GetDeclaredSymbol(attributedDeclaration);
+            }
+
+            if (attributedSymbol is null)
+            {
+                data = null;
+                return false;
+            }
+
+            var attributes = attributedSymbol.GetAttributes();
+            foreach (var attribute in attributes)
+            {
+                if (attribute.ApplicationSyntaxReference?.GetSyntax() == boundAttribute.Syntax)
+                {
+                    data = attribute;
+                    return true;
+                }
+            }
+
+            data = null;
+            return false;
+        }
+
+        private IOperation CreateBoundAttributeOperation(BoundAttribute boundAttribute)
+        {
+            var isImplicit = boundAttribute.WasCompilerGenerated;
+            if (!TryGetAttributeData(boundAttribute, out var attributeData))
+            {
+                var invalidOperation = OperationFactory.CreateInvalidOperation(_semanticModel, boundAttribute.Syntax, GetIOperationChildren(boundAttribute), isImplicit: true);
+                return new AttributeOperation(invalidOperation, _semanticModel, boundAttribute.Syntax, isImplicit);
+            }
+
+            var builder = ImmutableArray.CreateBuilder<IArgumentOperation>(boundAttribute.Constructor!.ParameterCount);
+            var seenParameters = ImmutableHashSet.CreateBuilder<ParameterSymbol>();
+
+            for (int i = 0; i < boundAttribute.ConstructorArguments.Length; i++)
+            {
+                int parameterIndex = boundAttribute.ConstructorArgumentsToParamsOpt.IsDefaultOrEmpty switch
+                {
+                    true => i,
+                    false => boundAttribute.ConstructorArgumentsToParamsOpt[i],
+                };
+
+                var parameter = boundAttribute.Constructor.Parameters[parameterIndex];
+                seenParameters.Add(parameter);
+                builder.Add(CreateArgumentOperation(ArgumentKind.Explicit, parameter.GetPublicSymbol(), boundAttribute.ConstructorArguments[i]));
+            }
+
+            foreach (var parameter in boundAttribute.Constructor.Parameters)
+            {
+                if (!seenParameters.Contains(parameter))
+                {
+                    var typedConstant = attributeData.CommonConstructorArguments[parameter.Ordinal];
+                    builder.Add(CreateArgumentOperation(ArgumentKind.DefaultValue, parameter.GetPublicSymbol(), new BoundLiteral(boundAttribute.Syntax, ConstantValue.Create(typedConstant.Value!, typedConstant.Type!.SpecialType), parameter.Type) { WasCompilerGenerated = true }));
```

Comment:
> new BoundLiteral

This is too fragile and doesn't handle conversions correctly 😕

Reply:
@AlekseyTs @RikkiGibson Any suggestions on the "proper" way of creating the argument operation when we don't have a `BoundExpression` node for it?